WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,18 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    self.rounds.each(&:destroy!)
+    self.reload
+    self.rounds_attributes = wcif["rounds"].map.with_index do |wcif_round, index|
+      round_number = index + 1
+      Round.wcif_to_round_attributes(wcif_round, round_number)
```

Comment:
`def rounds_attributes=` is the method created by [`accepts_nested_attributes_for`](https://github.com/jfly/worldcubeassociation.org/blob/c470fe4ce7ed79f0ba343888fb443233248bad97/WcaOnRails/app/models/competition_event.rb#L8).

I've made ths switch to do `self.rounds <<` as you suggest.